plugins/discovery-ec2/src/test/java/org/opensearch/discovery/ec2/Ec2DiscoveryTests.java

```diff
@@ -83,6 +82,12 @@ public class Ec2DiscoveryTests extends AbstractEC2MockAPITestCase {
 
     private Map<String, TransportAddress> poorMansDNS = new ConcurrentHashMap<>();
 
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        setUpAwsProfile();
```

Comment:
We need to reset properties to the previous values on `tearDown`

Reply:
Added AbstractEc2DiscoveryTestCase that does both setup and teardown.